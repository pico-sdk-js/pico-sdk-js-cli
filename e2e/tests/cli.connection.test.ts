import {describe, it} from '@jest/globals';
import psjRunner from './psjRunner';

describe('PSJ Connection Scenarios', () => {

    describe('command line args', () => {
        describe('--help', () => {
            it('shows help', async () => {
                await psjRunner()
                .start(['--help'])
                .assertExitCode(0)
                .assertSnapshot();     
            });
        });

        describe('--auto-connect', () => {
            it('connects to /dev/ttyACM0', async () => {
                await psjRunner()
                .start(['--auto-connect', '--skip-header'])
                .assertSnapshot();
            });
        });
    });

    describe('.exit', () => {
        it('in REPL quits process', async () => {
            await psjRunner()
            .start(['--skip-header'])
            .command('.exit')
            .assertSnapshot()
            .assertExitCode(0);
        });
    });

    describe('.ls', () => {
        it('shows error when not connected', async () => {
            await psjRunner()
            .start(['--skip-header'])
            .command('.ls')
            .assertSnapshot();
        });
    });

    describe('.connect', () => {
        it('connects to /dev/ttyACM0', async () => {
            await psjRunner()
                .start(['--skip-header'])
                .command('.connect')
                .assertSnapshot();
        });

        it('connection failure allows reconnect', async () => {
            await psjRunner()
                .start(['--skip-header'])
                .command('.connect -D /dev/ttyACM99')
                .command('.connect')
                .assertSnapshot();

        });
    });

});