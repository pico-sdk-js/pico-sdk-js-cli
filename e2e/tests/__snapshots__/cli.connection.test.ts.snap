// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PSJ Connection Scenarios .connect connection failure allows reconnect: stdOut 1`] = `
"> .connect -D /dev/ttyACM99
Connecting ... 
Connecting to serial device at /dev/ttyACM99
ERR: Error: No such file or directory, cannot open /dev/ttyACM99
> .connect
Connecting ... 
Connecting to serial device at /dev/ttyACM0
> "
`;

exports[`PSJ Connection Scenarios .connect connects to /dev/ttyACM0: stdOut 1`] = `
"> .connect
Connecting ... 
Connecting to serial device at /dev/ttyACM0
> "
`;

exports[`PSJ Connection Scenarios .exit in REPL quits process: stdOut 1`] = `"> .exit"`;

exports[`PSJ Connection Scenarios .ls shows error when not connected: stdOut 1`] = `
"> .ls
ERR: Not connected, run .connect to connect to a device running Pico-Sdk-JS.
> "
`;

exports[`PSJ Connection Scenarios command line args --auto-connect connects to /dev/ttyACM0: stdOut 1`] = `
"Connecting ... 
Connecting to serial device at /dev/ttyACM0
> "
`;

exports[`PSJ Connection Scenarios command line args --help shows help: stdOut 1`] = `
"psj
Starts the REPL server to connect to the Raspberry Pi Pico
Options:
      --help                Show help                                  [boolean]
      --version             Show version number                        [boolean]
      --skip-header         Do not output the process header.
                                                      [boolean] [default: false]
      --log-level, --ll     Sets the log level of the output.
     [string] [choices: "error", "warning", "debug", "trace"] [default: "error"]
      --auto-connect, --ac  Automatically connects on start.
                                                      [boolean] [default: false]
  -l, --local               Starts a local process to connect to. NOTE: Must set
                            the "PSJ_LOCAL" environment variable to the
                            pico-sdk-js executable.                    [boolean]
For more information, check out our docs on https://pico-sdk-js.github.io/"
`;
